src/java/org/apache/cassandra/index/sai/disk/oldlucene/LegacyByteBuffersDataInput.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.oldlucene;
+
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.RandomAccessInput;
+import org.apache.lucene.util.Accountable;
+import org.apache.lucene.util.RamUsageEstimator;
+
+/**
+ * A {@link DataInput} implementing {@link RandomAccessInput} and reading data from a
+ * list of {@link ByteBuffer}s. This uses the big-endian byte ordering of Lucene 7.5.
+ * Note that this participates in the type hierarchy of the modern Lucene depencency,
+ * so DataInput methods that assume little-endianness must be overridden.
+ */
+public final class LegacyByteBuffersDataInput extends DataInput implements Accountable, RandomAccessInput {
```

Comment:
potential subtle issue: One more thing that applies to this class and most other classes in this package – we're overriding some concrete methods and change their endianess here. If the base class gets enhanced with new public concrete methods later (e.g. after lucene version upgrade), the code would still compile, but those methods would be automatically inherited by the subclasses. And they might be incorrect, as the superclass uses little-endian encoding. I might be oversensitive here, but really a very similar thing already bit us in SAI some time before.

Having said that, I have no idea for creating a better design in this particular case, so the only idea that comes to my mind is - could we add a test that would verify if all methods from the base class are overriden? If we don't need to override, we could just override with plain delegate to super to explicitly mark it as "we handled this method, we are ok with the base class impl". In this case if a new method ever gets added to the base, the test would fail until it is taken care of by subclasses.